### PR TITLE
Limit

### PR DESCRIPTION
**Pull request by brettz9**
Builds on top of the promise-catching PR with one commit that does the following:

- Refactoring: Move `count` and `keys` next to `execute` so they can be separated from (less fundamental) query modifiers
- Refactoring: Alphabetize `Query` return objects for easier comparison;
- API fix: Disallow `map` on itself (only one will be used anyways);
- API change: Allow `desc`, `distinct`, `filter`, `keys`, `map`, `modify` on `limit` (part of #57);
- API change: Allow `limit` on `distinct`, `desc`, `keys`;
- README: Move `distinct` section so ranges section above it can be next to the section on `limit` ranges;
- README: Document that `filter` can be used multiple times

### CODE REVIEW

File: src/db.js [hunk 1]
```diff
@@ -204,82 +193,107 @@
...
                     filter,
-                    distinct,
+                    limit,
                     map
                 };
             };
-            const filter = function (...args) {
-                filters.push(args.slice(0, 2));
+
+            const limit = function (...args) {
```
Comment (aaronpowell) [inline src/db.js]: why would you use a rest parameter here, wouldn't it be simpler to just define `start` and `end`? Seems like ES6 for the sake of ES6

Comment (brettz9) [inline src/db.js]: Yes, agreed. I think I was just converting from `arguments` and hadn't taken it the step further as it should have, Fixed for `limit` and `filter` in the amended version I just rebased.

**Comment by brettz9** [comment]
Give me a little while to fix--seems failing for some reason...

File: src/db.js (hunk 1 above)
Comment (brettz9) [inline src/db.js]: Actually, upon further inspection, there actually had been a reason for the old way (it had checked for the length of the array, and if only 1 length, it added to it) but I've refactored `limit` to use named variables as well as `filter` which had similar checking.

**Pull request merged**